ngraph/src/ngraph/op/softplus.cpp

```diff
@@ -0,0 +1,84 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/itt.hpp"
+
+#include "ngraph/op/softplus.hpp"
+#include "ngraph/attribute_visitor.hpp"
+
+#include "ngraph/runtime/host_tensor.hpp"
+#include "ngraph/runtime/reference/softplus.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo op::v4::SoftPlus::type_info;
+
+op::v4::SoftPlus::SoftPlus(const Output<Node>& arg)
+    : Op({arg})
+{
+    constructor_validate_and_infer_types();
+}
+
+bool op::v4::SoftPlus::visit_attributes(AttributeVisitor& visitor)
+{
+    return true;
+}
+
+void op::v4::SoftPlus::validate_and_infer_types()
+{
+    set_output_size(1);
+    set_output_type(0, get_input_element_type(0), get_input_partial_shape(0));
+}
+
+shared_ptr<Node> op::v4::SoftPlus::clone_with_new_inputs(const OutputVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    return make_shared<op::v4::SoftPlus>(new_args.at(0));
+}
+
+namespace
+{
+    template <element::Type_t ET>
```

Comment:
Looks like we don't need template argument for this method.

We can get T type from `get_element_type`.

Please try to reduce the number of template code.